_includes/callouts.html

```diff
@@ -5,35 +5,78 @@
-                                <i class="fas {{ callout.icon }} fa-4x"></i>
-                            {% endif %}
-                            </div>
-                            {% endif %}
-                            <p class="title is-5">{{ callout.title }}</p>
-                            <p class="subtitle is-5">{{ callout.subtitle }}</p>
-                            
-                            {% if callout.description %}
-                            <div class="content">
-                                <p>{{ callout.description | newline_to_br }}</p>
-                            </div>
+                            <i class="fas {{ callout.icon }} fa-4x"></i>
                             {% endif %}
+                        </div>
+                        {% endif %}
+                        <p class="title is-5">
+                            <span id="countup_{{ forloop.index }}" data-final-count="{{ callout.title | remove: ',' }}">{{ callout.title }}</span>
+                        </p>
+                        <p class="subtitle is-5">{{ callout.subtitle }}</p>
 
-                            {% if callout.call_to_action_name %}
-                            <a href="{{ callout.call_to_action_link | relative_url }}" class="button is-primary">
-                                {{ callout.call_to_action_name }}
-                            </a>
-                            {% endif %}
+                        {% if callout.description %}
+                        <div class="content">
+                            <p>{{ callout.description | newline_to_br }}</p>
                         </div>
+                        {% endif %}
+
+                        {% if callout.call_to_action_name %}
+                        <a href="{{ callout.call_to_action_link | relative_url }}" class="button is-primary">
+                            {{ callout.call_to_action_name }}
+                        </a>
+                        {% endif %}
+                    </div>
                     {% endfor %}
                 </div>
             </div>
         </div>
     </section>
+
+<script>
+    const observerOptions = {
+        root: null,
+        rootMargin: '0px',
+        threshold: 0.5, // The element will start counting when at least 50% visible
+    };
+
+    document.addEventListener('DOMContentLoaded', function () {
+        const countUpElements = document.querySelectorAll('[data-final-count]');
+
+        const startCountUpAnimation = (entries, observer) => {
+            entries.forEach((entry) => {
+                if (entry.isIntersecting) {
+                    const element = entry.target;
+                    const finalCount = parseInt(element.getAttribute('data-final-count').replace(/,/g, ''));
+                    const minDuration = 2500; // 2.5 seconds
+                    const maxDuration = 5000; // 5 seconds
+                    const randomDuration = Math.random() * (maxDuration - minDuration) + minDuration;
```

Comment:
Does this need to happen each iteration or can it happen once for all iterations?
Is it OK if this is a decimal or does it need to be a whole number?

Reply:
The final count and random duration values should be different for every element, the duration values can be moved. https://github.com/FRCTeam1987/teambroncobots.com/commit/20b10936ff3fc738bd77f749d1803b615dc3bf18